ethcore/src/state/mod.rs

```diff
@@ -310,6 +311,11 @@ impl<B: Backend> State<B> {
 		Ok(state)
 	}
 
+	/// Get a VM factory that can execute on this state.
+	pub fn vm_factory(&self) -> EvmFactory {
```

Comment:
what was the motivation to move vm factory in the state?
how the decision of choosing the vm executor depends on state?